packages/clay-date-picker/src/index.tsx

```diff
@@ -319,24 +430,46 @@ const ClayDatePicker: React.FunctionComponent<IProps> = React.forwardRef<
 		 * is no date selected.
 		 */
 		const handleDotClicked = () => {
-			changeMonth(dateNow);
-			setDaySelected(dateNow);
-			onValueChange(dateNow);
+			changeMonth(NEW_DATE);
+
+			setDaysSelected([NEW_DATE, daysSelected[1]]);
+
+			if (range) {
+				return onValueChange(
+					fromRangeToString([NEW_DATE, daysSelected[1]], dateFormat)
+				);
+			}
+
+			if (time) {
+				const time = parseDate(currentTime, TIME_FORMAT, NEW_DATE);
+
+				return onValueChange(
+					formatDate(time, `${dateFormat} ${TIME_FORMAT}`)
+				);
+			}
+
+			const dateFormatted = formatDate(NEW_DATE, dateFormat);
+
+			return onValueChange(dateFormatted);
```

Comment:
The same here.

```js
onValueChange(
	range
		? fromRangeToString([DATE_NOW, daysSelected[1]], dateFormat)
		: time
			? formatDate(
				parseDate(currentTime, TIME_FORMAT, DATE_NOW),
				`${dateFormat} ${TIME_FORMAT}`
			)
			: formatDate(NEW_DATE, dateFormat)
);
```

If the ternary gets too weird, we still have other options to consider:

```js
if (range) {
	onValueChange(
		fromRangeToString([NEW_DATE, daysSelected[1]], dateFormat)
	);
} else if (time) {
	onValueChange(
		formatDate(
			parseDate(currentTime, TIME_FORMAT, NEW_DATE),
			`${dateFormat} ${TIME_FORMAT}`
		)
	);
} else {
	onValueChange(formatDate(NEW_DATE, dateFormat));
}
```

or (Particularly in that way I look a little weird yet)

```js
let dateFormatted;

if (range) {
	dateFormatted = fromRangeToString([NEW_DATE, daysSelected[1]], dateFormat);
} else if (time) {
	dateFormatted = formatDate(
		parseDate(currentTime, TIME_FORMAT, NEW_DATE),
		`${dateFormat} ${TIME_FORMAT}`
	)
} else {
	dateFormatted = formatDate(NEW_DATE, dateFormat);
}

onValueChange(dateFormatted);
```

@wincent any thoughts on that?

Reply:
Nested ternaries are obviously a bit harder to read than spelling it out with `if`/`else if`/`else`. I tend to reserve them for very specific cases (like in JSX, if the conditions are very simple, and brief enough not to wrap excessively).

So I like option 2 or 3. Option 2 is more verbose (😢 — more stuff to read), but option 3 has a `let` (😢 — more cognitive load because you have to look for mutation/assigment). I'd probably lean towards 3 for now, knowing that [once JS gets "do expressions"](https://github.com/tc39/proposal-do-expressions) you will be able to have the best of both worlds, perhaps (a `const` assignment plus DRY-ing up the calls to `onValueChange()`):

```js
const dateFormatted = do {
	if (range) {
		fromRangeToString([NEW_DATE, daysSelected[1]], dateFormat)
	} else if (time) {
		formatDate(
			parseDate(currentTime, TIME_FORMAT, NEW_DATE),
			`${dateFormat} ${TIME_FORMAT}`
		)
	} else {
		formatDate(NEW_DATE, dateFormat)
	}
};

onValueChange(dateFormatted);
```

Only stage 1 at this time though, so don't hold your breath.

Reply:
Yeah, the ternary here ended up degrading the reading here, sad... I'm also not a big fan of option 3 because it requires mutation in the variable but I wanted to avoid the repetitive function call.

the expression `do` could really help, I remember seeing it being used in JSX some time ago but it seems to have been stagnant 🤷‍♂️. Maybe in some parallel world, we can reduce this if we structure other parts of the component to converge in a simpler logical change 😄.